json/codec.go

```diff
@@ -20,7 +20,10 @@ type codec struct {
 }
 
 type encoder struct{ flags AppendFlags }
-type decoder struct{ flags ParseFlags }
+type decoder struct {
+	flags      ParseFlags
+	inputFlags inputFlags
```

Comment:
I remember observing that increasing the size of this struct had an impact on throughput. It may be interesting to try packing both the parse and input flags into a single `int` (maybe remove the `inputFlags` type and make the input flags unexported values of `ParseFlags`).

Reply:
See 237249f82543a557325ee6710ba4c0844045a1d0